### PR TITLE
feat(jans-lock): cedarling code refactoring to improve its readability and maintainance

### LINKED ISSUE #9484: refactor(cedarling): code refactoring to improve its readability and maintainance
```
src/
├── lib.rs
├── init/
│   ├── mod.rs
│   ├── submodule.rs
├── authz/
│   ├── mod.rs
│   └── another_submodule.rs
├── jwt/
│   ├── mod.rs
│   └── submodule.rs
├── log/
│   ├── mod.rs
│   └── submodule.rs
├── lock/
│    ├── mod.rs
│    └── submodule.rs
├── model/
│   ├── user.rs
│   ├── client.rs
|    ├── id_token.rs
│   └── acess_token.rs
├── test/
│    ├── mod.rs
├── services/
│    ├── mod.rs
```

**Other points**
the naming convention of each method, entity, variable should define its purpose.
comments should be used before functions , entities explaining what is is doing

### PR DESCRIPTION
**Pull request by olehbozhok**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

### Description
This PR includes skeleton of new version of Cedarling

#### Target issue
    [link](https://github.com/JanssenProject/jans/issues/9484)
  
    closes #9484

#### Implementation Details
  Code of Cedarling was removed. Created new skeleton.
  New code will be created according to the [Cedarling Build Plan](https://github.com/JanssenProject/jans/wiki/Cedarling-Build-Plan)

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [ ] Relevant unit and integration tests have been added/updated
- [x] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [ ] **I confirm that there is no impact on the docs due to the code changes in this PR.**

### CODE REVIEW

File: jans-lock/cedarling/rustfmt.toml [hunk 1]
```diff
@@ -1,2 +1,72 @@
-hard_tabs = true
+edition = "2021"
+unstable_features = true
+version = "Two"
```
Comment (nynymike) [inline jans-lock/cedarling/rustfmt.toml:3]: Version "Two"? What is the version of exactly? Why is it "Two"?

**Review by nynymike — CHANGES_REQUESTED** [review]
1. 
2. The license should actually read: 

```
/*
 * This software is available under the Apache-2.0 license. 
 * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
 *
 * Copyright (c) 2024, Gluu, Inc. 
 */
 ```
 2. Did you add `#![deny(missing_docs)]` ?

File: jans-lock/cedarling/rustfmt.toml (hunk 1 above)
Comment (olehbozhok) [reply jans-lock/cedarling/rustfmt.toml:3]: https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#version
here is link to the config documentation

Comment (olehbozhok) [reply jans-lock/cedarling/rustfmt.toml:3]:
```
Which version of the formatting rules to use. Version::One is backwards-compatible with Rustfmt 1.0. Other versions are only backwards compatible within a major version number.
```

https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#version
here is link to the config documentation

**Review by duttarnab — APPROVED** [review]

**Review by nynymike — APPROVED** [review]
Bueno!

**Review by moabu — APPROVED** [review]

**Pull request merged**